src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -466,349 +446,67 @@ Namespace Microsoft.CodeAnalysis.Semantics
-                            Case BinaryOperatorKind.NotEquals
-                                Return BinaryOperationKind.StringNotEquals
-                            Case BinaryOperatorKind.Like
-                                Return BinaryOperationKind.StringLike
-                        End Select
-                    Case SpecialType.System_Object
-                        Select Case operatorKind And BinaryOperatorKind.OpMask
-                            Case BinaryOperatorKind.Add
-                                Return BinaryOperationKind.ObjectAdd
-                            Case BinaryOperatorKind.Subtract
-                                Return BinaryOperationKind.ObjectSubtract
-                            Case BinaryOperatorKind.Multiply
-                                Return BinaryOperationKind.ObjectMultiply
-                            Case BinaryOperatorKind.Power
-                                Return BinaryOperationKind.ObjectPower
-                            Case BinaryOperatorKind.IntegerDivide
-                                Return BinaryOperationKind.ObjectIntegerDivide
-                            Case BinaryOperatorKind.Divide
-                                Return BinaryOperationKind.ObjectDivide
-                            Case BinaryOperatorKind.Modulo
-                                Return BinaryOperationKind.ObjectRemainder
-                            Case BinaryOperatorKind.Concatenate
-                                Return BinaryOperationKind.ObjectConcatenate
-                            Case BinaryOperatorKind.And
-                                Return BinaryOperationKind.ObjectAnd
-                            Case BinaryOperatorKind.Or
-                                Return BinaryOperationKind.ObjectOr
-                            Case BinaryOperatorKind.Xor
-                                Return BinaryOperationKind.ObjectExclusiveOr
-                            Case BinaryOperatorKind.AndAlso
-                                Return BinaryOperationKind.ObjectConditionalAnd
-                            Case BinaryOperatorKind.OrElse
-                                Return BinaryOperationKind.ObjectConditionalOr
-                            Case BinaryOperatorKind.LeftShift
-                                Return BinaryOperationKind.ObjectLeftShift
-                            Case BinaryOperatorKind.RightShift
-                                Return BinaryOperationKind.ObjectRightShift
-                            Case BinaryOperatorKind.LessThan
-                                Return BinaryOperationKind.ObjectLessThan
-                            Case BinaryOperatorKind.LessThanOrEqual
-                                Return BinaryOperationKind.ObjectLessThanOrEqual
-                            Case BinaryOperatorKind.Equals
-                                Return BinaryOperationKind.ObjectVBEquals
-                            Case BinaryOperatorKind.Is
```

Comment:
It would be good to have  a test for this operator and string operands

---
In reply to: [133323581](https://github.com/dotnet/roslyn/pull/21407#discussion_r133323581) [](ancestors = 133323581)